File: src/ice/world/consumers/ConsumePowerMultiple.java
package ice.world.consumers;

import ice.world.blocks.factoryBlocks.multipleCrafter.MultipleCrafter;
import mindustry.gen.Building;
import mindustry.world.Block;
import mindustry.world.consumers.ConsumePower;
import mindustry.world.meta.Stats;

public class ConsumePowerMultiple extends ConsumePower {
    public ConsumePower[] cons;

    public ConsumePowerMultiple(ConsumePower[] consumePowers) {
        cons = consumePowers;
    }

    @Override
    public void apply(Block block) {
    }

    @Override
    public boolean ignore() {
        return true;
    }

    @Override
    public void display(Stats stats) {
    }

    @Override
    public float requestedPower(Building entity) {
        if (entity instanceof MultipleCrafter.MultipleCrafterBuilding b) {
            return b.consPower != null ? b.consPower.requestedPower(b) : 0f;
        }
        return 0;
    }
}

File: src/ice/library/meta/consumers/ConsumePowerMultiple.kt
package ice.library.meta.consumers

import ice.library.baseContent.blocks.crafting.multipleCrafter.MultipleCrafter.MultipleCrafterBuilding
import mindustry.gen.Building
import mindustry.world.Block
import mindustry.world.consumers.ConsumePower
import mindustry.world.meta.Stats

class ConsumePowerMultiple(var cons: Array<ConsumePower>) : ConsumePower() {
    override fun apply(block: Block) {
    }

    override fun ignore(): Boolean {
        return true
    }

    override fun display(stats: Stats) {
    }

    override fun requestedPower(entity: Building): Float {
        if (entity is MultipleCrafterBuilding) {
            return if (entity.consPower != null) entity.consPower!!.requestedPower(entity) else 0f
        }
        return 0f
    }
}
